src/main/java/net/mcreator/ui/laf/MCreatorTheme.java

```diff
@@ -72,29 +75,17 @@ public Color getMainTint() {
 		return MAIN_TINT;
 	}
 
-	@Nonnull protected abstract Color getBlackAccent();
-
-	@Nonnull protected abstract Color getDarkAccent();
-
-	@Nonnull protected abstract Color getLightAccent();
-
-	@Nonnull protected abstract Color getGrayColor();
-
-	@Nonnull protected abstract Color getBrightColor();
-
-	@Nonnull protected abstract String getBlocklyCSSName();
-
-	@Nonnull protected abstract String getCodeEditorXML();
+	public ColorScheme getColorScheme() {
+		return colorScheme;
+	}
 
 	protected void initMCreatorThemeColors(UIDefaults table) {
-		table.put("MCreatorLAF.BLACK_ACCENT", getBlackAccent());
-		table.put("MCreatorLAF.DARK_ACCENT", getDarkAccent());
-		table.put("MCreatorLAF.LIGHT_ACCENT", getLightAccent());
-		table.put("MCreatorLAF.GRAY_COLOR", getGrayColor());
-		table.put("MCreatorLAF.BRIGHT_COLOR", getBrightColor());
+		table.put("MCreatorLAF.BLACK_ACCENT", colorScheme.getAltBackgroundColor());
+		table.put("MCreatorLAF.DARK_ACCENT", colorScheme.getBackgroundColor());
+		table.put("MCreatorLAF.LIGHT_ACCENT", colorScheme.getSecondAltBackgroundColor());

```

Comment:
You should swap secondalt and alt background color names in colorScheme, as LIGHT_ACCENT is used much more and is thus not really second, also reduces code bulkiness